numba/targets/boxing.py

```diff
@@ -310,6 +310,33 @@ def box_array(typ, val, c):
         c.pyapi.incref(parent)
         return parent
 
+@box(types.SmartArrayType)
+def box_smart_array(type, value, c):
+    # First build a Numpy array object, then wrap it in a Array
+    a = cgutils.create_struct_proxy(type)(c.context, c.builder, value=value)
+    # if 'parent' is set, we are re-boxing an object, so use the same logic
+    # as reflect.
+    obj = a.parent
+    res = cgutils.alloca_once_value(c.builder, obj)
+    with c.builder.if_else(cgutils.is_not_null(c.builder, obj)) as (has_parent, otherwise):
+        with has_parent:
+            c.pyapi.incref(obj)
+            c.pyapi.call_method(obj, "host_changed")
```

Comment:
I have attempted to do that with https://github.com/stefanseefeld/numba/commit/bf17e17a3184e43b9484c51a7bb7e870ad61c855, but that results in compilation errors

   ```
RuntimeError: Failed at nopython (nopython mode backend)
LLVM IR parsing error
<string>:74:1: error: expected instruction opcode
entry.endif.endif.endif.endif:
```

Any help with this would be much appreciated.